packages/microcosm/src/microcosm.js

```diff
@@ -496,6 +488,34 @@ class Microcosm extends Emitter implements Domain {
       this.actions[name] = tag(actions[name], name)
     }
   }
+
+  _enableDomains() {
+    if (this.domains) {
+      return
+    }
+
+    this.domains = new DomainEngine(this)
+
+    // When an action is first created
+    this.history.on('append', this._createSnapshot, this)
+
+    // When an action snapshot needs updating
+    this.history.on('update', this._updateSnapshot, this)
+
+    // When an action snapshot should be removed
+    this.history.on('remove', this._removeSnapshot, this)
+  }
```

Comment:
This here (avoiding snapshots for Microcosms that have no domains) is what made things so much faster.